src/experiment_prototype/experiment_prototype.py

```diff
@@ -433,26 +434,27 @@ def __init__(self, cpid, output_rx_rate=default_output_rx_rate, rx_bandwidth=def
         # Receiving is possible in the range of rxctrfreq +/- (rxrate/2)
 
         if self.txrate > self.options.max_tx_sample_rate:
-            errmsg = "Experiment's transmit bandwidth is too large: {} greater than max " \
-                     "{}.".format(self.txrate, self.options.max_tx_sample_rate)
+            errmsg = f"Experiment's transmit bandwidth is too large: {self.txrate} greater than " \
+                     f"max {self.options.max_tx_sample_rate}."
             raise ExperimentException(errmsg)
 
         if self.rxrate > self.options.max_rx_sample_rate:
-            errmsg = "Experiment's receive bandwidth is too large: {} greater than max " \
-                     "{}.".format(self.rxrate, self.options.max_rx_sample_rate)
+            errmsg = f"Experiment's receive bandwidth is too large: {self.rxrate} greater than " \
+                     f"max {self.options.max_rx_sample_rate}."
             raise ExperimentException(errmsg)
 
         if round(self.options.usrp_master_clock_rate / self.txrate, 3) % 2.0 != 0.0:
-            errmsg = "Experiment's transmit bandwidth {} is not possible as it must be an integer divisor of" \
-                     " USRP master clock rate {}".format(self.txrate, self.options.usrp_master_clock_rate)
+            errmsg = f"Experiment's transmit bandwidth {self.txrate} is not possible as it must be an " \
+                     f"integer divisor of USRP master clock rate {self.options.usrp_master_clock_rate}"
             raise ExperimentException(errmsg)
 
         if round(self.options.usrp_master_clock_rate / self.rxrate, 3) % 2.0 != 0.0:
-            errmsg = "Experiment's receive bandwidth {} is not possible as it must be an integer divisor of" \
-                     " USRP master clock rate {}".format(self.rxrate, self.options.usrp_master_clock_rate)
+            errmsg = f"Experiment's receive bandwidth {self.rxrate} is not possible as it must be an " \
+                     f"integer divisor of USRP master clock rate {self.options.usrp_master_clock_rate}"
             raise ExperimentException(errmsg)
 
-        self.__decimation_scheme = decimation_scheme
+        if decimation_scheme is None:
+            self.__decimation_scheme = create_default_scheme()
```

Comment:
```suggestion
        if decimation_scheme is None:
            decimation_scheme = create_default_scheme()
            self.__decimation_scheme = decimation_scheme
```
`decimation scheme` is referenced below, so can't be None